packages/amplify-cli/src/utils/run-build-command.ts

```diff
@@ -0,0 +1,29 @@
+import { $TSContext, AmplifyError, spinner } from 'amplify-cli-core';
+import { printer } from 'amplify-prompts';
+import { sync as execaSync } from 'execa';
+
+export default async function runBuildCommand(context: $TSContext) {
+  if (
+    context.exeInfo.inputParams.runBuildCommand &&
+    context.exeInfo.projectConfig.frontend === 'javascript' &&
+    context.exeInfo.projectConfig?.javascript?.config?.BuildCommand
+  ) {
+    const { BuildCommand } = context.exeInfo.projectConfig.javascript.config;
+    const buildCommandArray = BuildCommand.split(' ');
```

Comment:
what if a build script is something like `npx tsc -p "this path/has spaces/tsconfig.json"`?